### PR TITLE
Add OSX build guide

### PR DESCRIPTION
**Pull request by liminzhu**
Add OSX build guide based on https://obastemur.com/2016/07/18/how-to-compile-chakracore-on-macososx/

### CODE REVIEW

**Comment by liminzhu** [comment]
@obastemur @digitalinfinity

**Comment by obastemur** [comment]
:shipit:

File: Building-ChakraCore.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ## Windows ##
  
-You can build ChakraCore on Windows 7 SP1 or above, and Windows Server 2008 R2 or above, with either Visual Studio 2013 or 2015, as long as C++ support is installed[*](#build_for_arm). For Linux support, ChakraCore is making progress in the [Linux branch](https://github.com/Microsoft/ChakraCore/tree/linux). Refer to the [Linux](#linux) section for building instructions on Linux.  
+You can build ChakraCore on Windows 7 SP1 or above, and Windows Server 2008 R2 or above, with either Visual Studio 2013 or 2015, as long as C++ support is installed[*](#build_for_arm). For Linux/OS X support, ChakraCore is making progress in the [Linux branch](https://github.com/Microsoft/ChakraCore/tree/linux). Refer to the [Linux](#linux) and [OS X](#os-x) sections for building instructions on these platforms.
```
Comment (dilijev) [inline Building-ChakraCore.md:4]: The sentence seems a little awkward to me with the extra platform in there. How about this?

"Refer to the [Linux](#linux) and [OS X](#os-x) sections for instructions for building on these platforms."

File: Building-ChakraCore.md [hunk 2]
```diff
@@ -69,3 +69,24 @@ To build ChakraCore on Linux:
 	* ```sudo apt-get install -y meld```
 	* Download and install [VSCode](https://code.visualstudio.com/Docs/editor/setup#_linux).
 	
+## OS X ##
+
+Similar to [Linux](#Linux), we have experimental implementation of ChakraCore interpreter and runtime without JIT for OS X 10.9+. Please make sure you also have XCode and [Homebrew](http://brew.sh/).
```
Comment (dilijev) [inline Building-ChakraCore.md]: > we have experimental implementation

Maybe add "an":

> "We have an experimental implementation"

**Comment by dilijev** [comment]
Otherwise LGTM

**Comment by liminzhu** [comment]
Thanks @dilijev ! How do merge the change? Just click the Merge PR button?

**Comment by dilijev** [comment]
@liminzhu I would like to avoid merges in the wiki history when possible, so I've been following these steps:
- Set up remotes if you haven't already. I like this:
  - `git remote add wiki https://github.com/Microsoft/ChakraCore.wiki.git` -- the ChakraCore/wiki page
  - `git remote add repo https://github.com/Microsoft/ChakraCore-wiki.git` -- the ChakraCore-wiki repo
- Switch to branch `master` if you aren't already on it.
- `git fetch wiki`
- `git merge --ff-only wiki/master`
- `git push repo master`
  - Assume there will be no conflicts if the ChakraCore-wiki repo is strictly behind (i.e. nobody has merged into this repo without then pushing the changes to the wiki. Otherwise a merge will be necessary.
- Use the green button to select the squash option (to prevent merges in the history).
- `git fetch repo`
- `git merge --ff-only repo/master`
- `git push wiki master`

Basically that's the long-winded variety of the pull-push-sync workflow.
